src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Tool\Bundle\MeasureBundle\Controller\InternalApi;
+
+use Akeneo\Tool\Bundle\MeasureBundle\Application\SaveMeasurementFamily\SaveMeasurementFamilyCommand;
+use Akeneo\Tool\Bundle\MeasureBundle\Application\SaveMeasurementFamily\SaveMeasurementFamilyHandler;
+use Akeneo\Tool\Bundle\MeasureBundle\Controller\ExternalApi\JsonSchemaErrorsFormatter;
+use Akeneo\Tool\Bundle\MeasureBundle\Controller\InternalApi\JsonSchema\MeasurementFamilyStructureValidator;
+use Akeneo\Tool\Component\Api\Exception\ViolationHttpException;
+use Akeneo\Tool\Component\Api\Normalizer\Exception\ViolationNormalizer;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class CreateMeasurementFamilyAction
+{
+    /** @var MeasurementFamilyStructureValidator */
+    private $measurementFamilyStructureValidator;
+
+    /** @var ValidatorInterface */
+    private $validator;
+
+    /** @var ViolationNormalizer */
+    private $violationNormalizer;
+
+    /** @var SaveMeasurementFamilyHandler */
+    private $saveMeasurementFamilyHandler;
+
+    public function __construct(
+        MeasurementFamilyStructureValidator $measurementFamilyStructureValidator,
+        ValidatorInterface $validator,
+        ViolationNormalizer $violationNormalizer,
+        SaveMeasurementFamilyHandler $saveMeasurementFamilyHandler
+    ) {
+        $this->measurementFamilyStructureValidator = $measurementFamilyStructureValidator;
+        $this->validator = $validator;
+        $this->violationNormalizer = $violationNormalizer;
+        $this->saveMeasurementFamilyHandler = $saveMeasurementFamilyHandler;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        if (!$request->isXmlHttpRequest()) {
+            return new RedirectResponse('/');
+        }
+
+        $normalizedRequest = $this->normalizeRequest($request);
```

Comment:
```suggestion
        $normalizedRequest = $this->decodeRequest($request);
```

I know you copy and pasted this from asset/RE but this is the correct naming.

FYI: 
JSON string -> [decode] -> php array -> [denormalize] -> php object -> [normalize] -> php array -> [encode] -> JSON string

[decode] + [denormalize] = deserialize
[normalize] + [encode] = serialize 

:)

Reply:
cc @SamirBoulil @ValentinMumble :)

Reply:
It's not even from asset/RE, I've kind of followed the ExternalApi from the MeasureBundle: https://github.com/akeneo/pim-community-dev/blob/cb0041544c45c9e74a92d5f49a4d514848b4c391/src/Akeneo/Tool/Bundle/MeasureBundle/Controller/ExternalApi/SaveMeasurementFamiliesAction.php#L94

Reply:
done